notebooks/Binary Classification with the UCI Credit Card Payment Default Dataset.ipynb

```diff
@@ -0,0 +1,599 @@
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "models_dict = {\"Unmitigated\": (test_preds, test_scores),\n",
+    "              \"ThresholdOptimizer\": (postprocess_preds, postprocess_preds)}\n",
+    "get_metrics_df(models_dict, Y_test, A_str_test)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We note that the `ThresholdOptimizer` method significantly reduces demographic disparity across multiple metrics. \n",
+    "\n",
+    "Below, we compare this model with the unmitigated `lightgbm` model using the `fairlearn` dashboard. "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**Gender fairness: Dashboard demo**"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "FairlearnDashboard(sensitive_features=A_str_test, sensitive_feature_names=['Sex'],\n",
+    "                   y_true=Y_test,\n",
+    "                   y_pred={\"Unmitigated\": test_preds,\n",
+    "                          \"ThresholdOptimizer\": postprocess_preds})"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Mitigating Demographic Disparities with GridSearch"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We now attempt to mitigate disparities using the `GridSearch` algorithm from `fairlearn`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "# Train GridSearch\n",
+    "sweep = GridSearch(model,\n",
+    "                   constraints=EqualizedOdds(),\n",
+    "                   grid_size=41,\n",
+    "                   grid_limit=2)\n",
+    "\n",
+    "sweep.fit(df_train_balanced, Y_train_balanced, sensitive_features=A_train_balanced)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "predictors = [result.predictor for result in sweep.all_results]\n",
+    "sweep_preds = [predictor.predict(df_test) for predictor in predictors] \n",
+    "sweep_scores = [predictor.predict_proba(df_test)[:, 1] for predictor in predictors] "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "equalized_odds_sweep = [\n",
+    "    group_equalized_odds_diff(Y_test, preds, A_str_test)\n",
+    "    for preds in sweep_preds\n",
+    "]\n",
+    "balanced_accuracy_sweep = [balanced_accuracy_score(Y_test, preds) for preds in sweep_preds]\n",
+    "auc_sweep = [roc_auc_score(Y_test, scores) for scores in sweep_scores]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Select only non-dominated models from the gridsearch\n",
```

Comment:
I'm guilty of this in my notebooks, but can you note why a model would be considered 'dominated' in the search?